rotkehlchen/db/history_events.py

```diff
@@ -861,11 +861,13 @@ def query_wrap_stats(self, from_ts: Timestamp, to_ts: Timestamp) -> dict[str, An
                 for symbol, amount in cursor
             ]
             cursor.execute(
-                "SELECT unixepoch(date(datetime(timestamp, 'unixepoch'), 'localtime'), 'utc'), COUNT(*) as tx_count FROM "  # noqa: E501
-                'evm_transactions WHERE timestamp >= ? AND timestamp <= ? '
-                "GROUP BY date(datetime(timestamp, 'unixepoch'), 'localtime') ORDER BY "
+                "SELECT unixepoch(date(datetime(timestamp/1000, 'unixepoch'), 'localtime'), 'utc'), COUNT(DISTINCT event_identifier) as tx_count "  # noqa: E501
+                'FROM evm_events_info JOIN history_events ON evm_events_info.identifier = history_events.identifier '  # noqa: E501
+                'WHERE timestamp >= ? AND timestamp <= ? AND history_events.asset NOT IN '
+                "(SELECT value FROM multisettings WHERE name = 'ignored_asset') "
+                "GROUP BY date(datetime(timestamp/1000, 'unixepoch'), 'localtime') ORDER BY "
                 'tx_count DESC LIMIT 10',
```

Comment:
SEems that something is still off. 
![image](https://github.com/user-attachments/assets/a4458bd7-a1e7-4a9d-b12a-f7defd985df4)

This is what it show to me but If I go to the history events and filter
![image](https://github.com/user-attachments/assets/c8ecab88-634a-4aca-9b97-f3e54c53bf2e)

if Instead I include the ignored assets 
![image](https://github.com/user-attachments/assets/9ae114e3-fcf5-4df3-9ff1-b5a3f674d435)

the number matches. I thought that also one case if for transactions that have more than one event with spam assets. Is that one handled?

Reply:
Transactions per chain should skip spam txs now as well. As mentioned in discord I had only modified the query for days with most transactions.

> for transactions that have more than one event with spam assets. Is that one handled?

Its checking history_events with spam assets, so if a transaction has multiple events, it will still count it as long as there is one event with non spam asset, but if all the transaction's events have ignored assets then the entire transaction will be skipped.